src/time.cpp

```diff
@@ -43,12 +43,26 @@ namespace fc {
       return fc::time_point_sec( (pt - epoch).total_seconds() );
   } FC_RETHROW_EXCEPTIONS( warn, "unable to convert ISO-formatted string to fc::time_point_sec" ) }
 
-  time_point::operator fc::string()const
-  {
-      const auto ptime = boost::posix_time::from_time_t( time_t( sec_since_epoch() ) );
-      auto msec = (elapsed.count() % 1000000) / 1000 + 1000;
-      return boost::posix_time::to_iso_extended_string( ptime ) + "."+to_string(msec).substr(1);
-  }
+   time_point::operator fc::string()const
+   {
+      auto padded_ms = [](uint64_t ms) {
+         uint64_t offset_ms = ms + 1000ULL;
+         return to_string(offset_ms).substr(1);
+      };
+
+      auto count = elapsed.count();
+      if (count >= 0) {
+         uint64_t secs = (uint64_t)count / 1000000ULL;
+         uint64_t msec = ((uint64_t)count % 1000000ULL) / 1000ULL;
+         const auto ptime = boost::posix_time::from_time_t(time_t(secs));
+         return boost::posix_time::to_iso_extended_string(ptime) + "." + padded_ms(msec);
+      } else {
+         // negative time_points are non-sensical but expressible with the current constraints
+         uint64_t secs = (uint64_t)(-count) / 1000000ULL;
+         uint64_t msec = ((uint64_t)(-count) % 1000000ULL) / 1000ULL;
+         return string("-") + to_string(secs) + "." + padded_ms(msec) + "s";
```

Comment:
Appends an `"s"` to the end of negative time point. All the examples I see with boost posix time and iso specs, none include a trailing `s`.